src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs

```diff
@@ -131,13 +132,40 @@ protected override Items.IItemsViewSource CreateItemsViewSource()
 		private protected override async void AttachingToWindow()
 		{
 			base.AttachingToWindow();
+			// Refresh the current visible item to catch any ItemsSource changes that occurred on other pages
+			// This ensures that updates made on other pages are reflected when navigating back
+			if (_wasDetachedFromWindow)
+			{
+				RefreshVisibleItems();
+			}
 			Setup(ItemsView);
+			_wasDetachedFromWindow = false;
 			// if we navigate back on NavigationController LayoutSubviews might not fire.
 			await UpdateInitialPosition();
 		}
 
+		void RefreshVisibleItems()
+		{
+			if (CollectionView is null || ItemsSource?.ItemCount == 0)
+			{
+				return;
+			}
+			// Get current visible item index paths to ensure proper refresh of carousel items
+			// Create a defensive copy to avoid issues if ItemsSource changes during execution
+			var indexPaths = CollectionView.IndexPathsForVisibleItems;
+			if (indexPaths?.Length > 0)
+			{
+				// Create a copy of the array to prevent concurrent modification issues
+				var indexPathsCopy = new NSIndexPath[indexPaths.Length];
+				Array.Copy(indexPaths, indexPathsCopy, indexPaths.Length);
```

Comment:
The ObjC binding is copying the array to managed code already? It looks something like:

```csharp
CFArray.ArrayFromHandle<NSIndexPath>(Messaging.NativeHandle_objc_msgSend(base.Handle, Selector.GetHandle("indexPathsForVisibleItems")), releaseHandle: false);
```

We shouldn't need to copy it a second time in managed code?